akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -56,15 +53,15 @@ private[remote] final class InboundCompressionsImpl(
   inboundContext: InboundContext,
   settings:       ArterySettings.Compression) extends InboundCompressions {
 
-  private val stopped = new AtomicBoolean
-
   // None is used as tombstone value after closed
-  // TOOD would be nice if we can cleanup the tombstones
+  // TODO would be nice if we can cleanup the tombstones
+  // FIXME we should be able to remove the tombstones easily now
   private[this] val _actorRefsIns = new Long2ObjectHashMap[Option[InboundActorRefCompression]]()
   private val createInboundActorRefsForOrigin = new LongFunction[Option[InboundActorRefCompression]] {
     override def apply(originUid: Long): Option[InboundActorRefCompression] = {
       val actorRefHitters = new TopHeavyHitters[ActorRef](settings.ActorRefs.Max)
-      Some(new InboundActorRefCompression(system, settings, originUid, inboundContext, actorRefHitters, stopped))
+      val log = Logging(system, classOf[InboundActorRefCompression])
```

Comment:
has to be reated here from the system anyway, once per creation of `InboundCompressionsImpl` not sure we can work around that?